### PR TITLE
PMA-024(Refactor)

### PR DESCRIPTION
**Pull request by AlexanderSUS**
1. Refactor reducers
2. Little fixes

### CODE REVIEW

File: src/store/authSlice.ts [hunk 1]
```diff
@@ -11,6 +13,12 @@ import { initialState, TOKEN } from '../constants/authorization';
 import { AuthState, JwtData } from '../types/authTypes';
 import UserService from '../api/userServise';
 
+type GenericAsyncThunk = AsyncThunk<SignInResponse | SignUpResponse, UserDataParams | string | User,
+{ state: RootState, rejectWithvalue: ValidationErrors }>;
```
Comment (AnAtoliy-AA) [inline src/store/authSlice.ts:17]: you can add type for this string

File: src/store/authSlice.ts [hunk 2]
```diff
@@ -11,6 +13,12 @@ import { initialState, TOKEN } from '../constants/authorization';
...
+
+type PendingAction = ReturnType<GenericAsyncThunk['pending']>;
+type RejectedAction = ReturnType<GenericAsyncThunk['rejected']>;
```
Comment (AnAtoliy-AA) [inline src/store/authSlice.ts:20]: add this values to enum `'rejected'`

File: src/store/boardSlice.ts [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
 import initialState from '../constants/boards';
 import type { ModalInputData } from '../types/modal';
 
-export const getBoards = createAsyncThunk<Boards, VoidArgument, {
+type GenericAsyncThunk = AsyncThunk<Boards, null | ModalInputData,
+{ state: RootState, rejectWithvalue: ValidationErrors }>;
+
+type PendingAction = ReturnType<GenericAsyncThunk['pending']>;
+type RejectedAction = ReturnType<GenericAsyncThunk['rejected']>;
+type FulfilledAction = ReturnType<GenericAsyncThunk['fulfilled']>;
```
Comment (AnAtoliy-AA) [inline src/store/boardSlice.ts:17]: the same as upper, maybe not enum, but consts

File: src/store/boardSlice.ts [hunk 4]
```diff
@@ -88,74 +99,38 @@ const boardSlice = createSlice({
...
+      (action): action is RejectedAction => action.type.endsWith('/rejected'),
+      (state, action) => {
+        if (isARequestedAction(action)) {
+          state.pending = false;
+          if (action.payload) {
+            const error = action.payload as ErrorResponseData;
+            state.error = error.message;
+            return;
+          }
+          state.error = 'Server error';
```
Comment (AnAtoliy-AA) [inline src/store/boardSlice.ts:121]: this values also to const

File: src/store/columnSlice.ts [hunk 5]
```diff
@@ -1,17 +1,25 @@
...
 import initialState from '../constants/columns';
 
-export const getColumns = createAsyncThunk<Column[], VoidArgument, {
-  state: RootState, rejectWithValue: ValidationErrors } >(
+type GenericAsyncThunk = AsyncThunk<Column[], null | ModalInputData,
+{ state: RootState, rejectWithvalue: ValidationErrors }>;
+
+type PendingAction = ReturnType<GenericAsyncThunk['pending']>;
+type RejectedAction = ReturnType<GenericAsyncThunk['rejected']>;
+type FulfilledAction = ReturnType<GenericAsyncThunk['fulfilled']>;
```
Comment (AnAtoliy-AA) [inline src/store/columnSlice.ts:17]: the same as upper

File: src/pages/Registration.tsx [hunk 6]
```diff
@@ -23,9 +23,15 @@ const Registration: React.FC = () => {
     }
   }, [dispatch, newUser]);
 
+  // TODO find solution to fix eslint error,
+  // DO NOT PUT error.message in dependency!!!!
+  // it cause ERRASE the error MESSAGE  adnd
+  // excessive dispatch call
```
Comment (AnAtoliy-AA) [inline src/pages/Registration.tsx:29]: remove this TODO bug fixed

**Review by allyksander — APPROVED** [review]

**Review by Gibr — APPROVED** [review]

File: src/store/authSlice.ts (hunk 1 above)
Comment (AlexanderSUS) [reply src/store/authSlice.ts:17]: Ok, will do it in next branch

File: src/store/authSlice.ts (hunk 2 above)
Comment (AlexanderSUS) [reply src/store/authSlice.ts:20]: Will do it in next branch

File: src/store/boardSlice.ts (hunk 3 above)
Comment (AlexanderSUS) [reply src/store/boardSlice.ts:17]: Will do it in next branch

File: src/store/columnSlice.ts (hunk 5 above)
Comment (AlexanderSUS) [reply src/store/columnSlice.ts:17]: Will do it in next branch

File: src/pages/Registration.tsx (hunk 6 above)
Comment (AlexanderSUS) [reply src/pages/Registration.tsx:29]: Will do it in next branch

**Pull request merged**